plugins/in_dummy/in_dummy.c

```diff
@@ -103,14 +248,23 @@ static int in_dummy_collect(struct flb_input_instance *ins,
         return -1;
     }
 
-    if (ctx->fixed_timestamp == FLB_FALSE) {
-        msgpack_sbuffer_init(&mp_sbuf);
-        gen_msg(ins, in_context, &mp_sbuf);
-        flb_input_log_append(ins, NULL, 0, mp_sbuf.data, mp_sbuf.size);
-        msgpack_sbuffer_destroy(&mp_sbuf);
+    if (strcasecmp(ctx->event_type, "log") == 0) {
```

Comment:
It is better to call strcasecmp only on init function and save as an enum value.
```c
// on init
if (strcasecmp(ctx->event_type_str, "log") == 0) {
   ctx->event_type = EVENT_TYPE_LOG;
}
```

```c
// on collect
if (ctx->event_type == EVENT_TYPE_LOG) {
   ...
}
```